assets/src/components/ladder.tsx

```diff
@@ -39,38 +46,70 @@ const Ladder = ({
   ladderDirection,
   selectedVehicleId,
 }: Props) => {
-  const dispatch = useContext(DispatchContext)
-  const timepointSpacingY =
-    (HEIGHT - MARGIN_TOP_BOTTOM * 2) / (timepoints.length - 1)
-
   const orderedTimepoints: Timepoint[] =
     // Use slice to make a copy of the array before destructively reversing
     ladderDirection === LadderDirection.OneToZero
       ? timepoints.slice().reverse()
       : timepoints
 
+  const timepointSpacingY: TimepointSpacingYFunc = (height: number): number =>
+    (height - MARGIN_TOP_BOTTOM * 2) / (timepoints.length - 1)
   const timepointStatusY = timepointStatusYFromTimepoints(
     timepoints,
     timepointSpacingY
   )
+
   const { ladderVehicles, widthOfLanes } = ladderVehiclesFromVehicles(
     vehicles,
     ladderDirection,
     timepointStatusY
   )
 
   const width = 120 + 2 * widthOfLanes
-  // (0, 0) is in the center of the first timepoint
-  const viewBox = [-width / 2, -MARGIN_TOP_BOTTOM, width, HEIGHT].join(" ")
 
   return (
-    <svg className="m-ladder" height={HEIGHT} width={width} viewBox={viewBox}>
+    <div className="m-ladder">
```

Comment:
Why does this need to go on a wrapping div? Can the `ref` and the class go together? If the class is on the `svg` directly, that's one fewer layer, and I think a `[0]` can be removed from `useLadderSvgHeight`.

Reply:
We need the div to have something rendered in the dom in order to pull a height off of. That's the height used in generating the svg.